### DISCUSSION TITLE
Color en checkboxes disabled

### DISCUSSION THREAD

**Discussion by aweell** [Telefonica/mistica-design#941] [Proposals]
Desde Movistar España llega el siguiente feedback:

> En los test de usuario están detectando que los usuarios no perciben el check deshabilitados como un estado deshabilitado. El problema que ven es que apenas se distingue del estado off (ya que es un grey3 que poco difiere cuando aplicas la norma del 50% de opacidad. Por tanto, por lo visto, a los usuarios no les parece que el check inactivo deshabilitado lo perciban como un estado deshabilitado. Quizás tengamos que darle una vuelta y ver opciones en las que esto parezca deshabilitado.
>  
> ![Screenshot 2022-12-05 at 13 06 51](https://user-images.githubusercontent.com/44420072/205633491-01900b9c-ea6f-4ebd-a1e4-ad20ba0bf505.png)
> 
> 
> La solución que recomiendan por aquí es o bien oscurecer el borde del estado off con un grey4 y cambiar el disabled a un grey2.
> 
> ![Screenshot 2022-12-05 at 13 07 50](https://user-images.githubusercontent.com/44420072/205633657-77586e48-2f1b-4994-8027-972a797edc0c.png)
>  
> O bien oscurecer el disabled con un fondo grey2
> 
> ![Screenshot 2022-12-05 at 13 08 16](https://user-images.githubusercontent.com/44420072/205633744-4e79b3c7-d215-4431-ba81-d60962583552.png)
>  
> El caso es que esto habría que aplicarlo al resto de controls. No sé si a vosotros se os ocurre una idea mejor.

**Comment by aweell** [comment]
### Oscurecer el border con grey 4

Ahora mismo tal y como está la implementación va a generarse un problema parecido, ya que al hacer el 50% del grey 4, aunque menos sutil, va a seguir sin existir una diferencia notable entre el enabled y el disabled.

<img width="755" alt="Screenshot 2022-12-13 at 13 58 47" src="https://user-images.githubusercontent.com/44420072/207324477-e856f17a-2e8f-4689-9b90-183f39ce9ea2.png">


### Uso de grey background

Actualmente por ejemplo estamos usando un grey background para indicar que el input es `readOnly` (derecha):

![Screenshot 2022-12-05 at 13 23 59](https://user-images.githubusercontent.com/44420072/205636761-bfc34dd4-b864-43db-833b-ec9a4a61feb8.png)

Podría generar inconsistencias que en los inputs se emplee el color de la misma manera para diferentes estados.

### Usar el `controlActivated` en borde de los unselected

Mi propuesta es que se utilice el `controlActivated` para el border del estado `enabledUnselected` y customizar tanto para `radioButton` como para `checkbox` los estados disabled, usando el `control` únicamente en esos estados:

<img width="838" alt="Screenshot 2022-12-13 at 13 57 32" src="https://user-images.githubusercontent.com/44420072/207324510-af929d08-7be3-473e-9e08-0ada8b30d06d.png">


**Nota:** esto tendría impacto en todas las marcas

**Comment by dsys-col** [comment]
Muy interesante lo que comentas. Me parece una buena solución separando lo que es clicable de lo que no lo es y que tal y como está ahora estamos viendo que es algo confuso para un usuario. Esto solamente lo extendería a los radiobuttons. Es fácil de modificar y aplicar a las diferentes marcas y el impacto no es excesivo y estaría limitado.

![image](https://user-images.githubusercontent.com/81299625/206173043-7e621013-99f1-46ed-a7c5-1c90725187af.png)

**Comment by yceballost** [comment]
Creada una issue para trabajar en profundidad sobre esto y traer alternativas a la discussion
https://github.com/Telefonica/mistica-design/issues/1011

**Comment by yceballost** [comment]
Se han hecho mejoras en el contraste de los bordes para pasar accesibilidad en cuanto al contraste de color y creemos que resuelve este problema

[Ejemplo](https://mistica-web.vercel.app/playroom/#?code=N4Igxg9gJgpiBcIA8BlALgQzAawAQGcAHLGAXmAA4BfAPgB0A7XXJAYQAsYcAjCAD1wMMAWzJ1wnHv3E0OXbLz5IA9HKl96TFmoX9cUAJb4M3ADYwogkWInzFMxs2Y77WlS-6bH2yboFhfCytRUnEAu2kQTSdcDz5vd19FaJ8I-0DLQ2MzIKEQsKTIlOdC%2BLdVUs0VdCxsTRAAGhA0TlF8BABtEABZCAA3I0wAJ0aQADUDPogAfQA5GAB3EABdJoWDKBb2%2BA6AZgA2AAYGgEYDimWqIA)
